pandas/core/base.py

```diff
@@ -1215,8 +1215,8 @@ def _map_values(self, mapper, na_action=None):
         # we can fastpath dict/Series to an efficient map
         # as we know that we are not going to have to yield
         # python types
-        if isinstance(mapper, dict):
-            if hasattr(mapper, "__missing__"):
+        if isinstance(mapper, abc.Mapping):
+            if isinstance(mapper, dict) and hasattr(mapper, "__missing__"):
```

Comment:
why do you think you need the if condition that was added?

Reply:
`collections.abc.Mapping` does not declare nor implement `__missing__`, that is a concept of `dict`. In case you use a subclass of `collections.abc.Mapping` which isn't a subclass of `dict`, you cannot rely on `__missing__`.
You can see an example on the second test I added, `test_map_abc_mapping_with_missing`.

Reply:
I think can just simplify to `isinstance(mapper, dict)` then no? Check for __missing__ after that is duplicative?

Reply:
No, since `dict` doesn't have a `__missing__` method on its own.
In `dict.__getitem__`'s implementation there is a reference to `__missing__` in case the object is a subclass of `dict` and implements it, but `dict` itself does not implement `__missing__`.
I suggest reading [dict's documentation](https://docs.python.org/3/library/stdtypes.html#dict), specifically the `__getitem__` part noted as `d[key]` for explanation.